lib/ValidationError.ts

```diff
@@ -18,18 +18,42 @@ export class ValidationError extends Error {
   }
 }
 
+function fromZodIssue(
+  issue: zod.ZodIssue,
+  issueSeparator: string,
+  unionSeparator: string
+): string {
+  if (issue.code === 'invalid_union') {
+    return issue.unionErrors
+      .map((zodError) =>
+        zodError.issues
+          .map((issue) => fromZodIssue(issue, issueSeparator, unionSeparator))
+          .join(issueSeparator)
+      )
+      .join(unionSeparator);
+  }
+
+  if (issue.path.length > 0) {
+    return `${issue.message} at "${joinPath(issue.path)}"`;
+  }
+
+  return issue.message;
+}
+
 export function fromZodError(
   zodError: zod.ZodError,
   options: {
     maxIssuesInMessage?: number;
     issueSeparator?: string;
+    unionSeparator?: string;
     prefixSeparator?: string;
     prefix?: string;
   } = {}
 ): ValidationError {
   const {
     maxIssuesInMessage = 99, // I've got 99 problems but the b$tch ain't one
     issueSeparator = '; ',
+    unionSeparator = ', or: ',
```

Comment:
issue: what if the user makes use of `zod.and()`. Wouldn't this return `or` by default? That doesn't seem right.

Reply:
ignore the above comment. I just checked that in the case of `zod.and()` the output is NOT a union,